cl/citations/utils.py

```diff
@@ -59,3 +59,15 @@ def is_balanced_html(text):
         return True
     except etree.XMLSyntaxError:
         return False
+
+
+def remove_duplicate_citations_by_regex(citations):
+    """Give a list of Citation objects, remove any that generate the
+    same regex. Because we each citation's regex matches every instance
+    of that citation in an opinion, we do not need to redundantly match
+    multiple citations with the same regexes."""
+    from cl.citations.models import Citation
+
+    Citation.__eq__ = Citation.regex_eq
+    Citation.__hash__ = Citation.regex_hash
```

Comment:
You know, I also had this question, I sort of hoped you would catch me on this... I admit I don't really know anything about the Python runtime, so I have no idea how "long" these changes last. Perhaps what would be safest would be to just reset to the original functions afterwards?

Reply:
I'm about 60% sure you're modifying the global state here, but if neither of us actually knows, that's a hint we shouldn't do it this way. Best case, it works but raises eyebrows every time somebody looks at it. 

I also wouldn't do it even temporarily b/c that could affect threaded stuff. We don't do much with threads, but I don't want to have questions about whether threads work if we ever go that route.

So...I think that leaves us with a more verbose approach of just calling the functions directly. Should be easy enough? 

(I also wouldn't rely on me to catch stuff like this unless you put a comment for me to comment on it during review. You're good enough that I often will just trust your work without reviewing it super carefully. If I merged it, you wouldn't know if it was because I missed it or b/c it was fine.)

Reply:
Haha, touché. I will refactor this.